codalab/apps/newsletter/views.py

```diff
@@ -25,74 +25,80 @@ def _send_mail(context_data, from_email=None, html_file=None, text_file=None, su
 
 
 def newsletter_signup(request):
-    form = NewsletterSubscriptionSignUpForm(request.POST or None)
+    if settings.MAILCHIMP_API_KEY:
```

Comment:
I don't think we need to do this check on these views, can just hide the ability to see/hit newsletter related stuff and have nice default behavior if no api key

Reply:
Added a default to 404 if no api key is set, otherwise we'll go to the correct form view.